string dataDir
string modelName
string detector
string descriptor
---
bool success